### PR TITLE
InDesign finalizations

### PR DESCRIPTION
**Pull request by rigaspapas**
Related to [TX-9144: InDesign's finalizations](https://transifex.atlassian.net/browse/TX-9144)

Checklist (for the reviewer)
----------------------------

* [ ] Problem and solution are well-explained in the PR
* [ ] Change is covered by unit-tests
* [ ] Code is well documented
* [ ] Code is styled well and is following best practices
* [ ] Performs well when applied to big organizations/resources/etc from our production database
* [ ] Errors are handled properly
* [ ] All (conceivable) edge-cases are handled
* [ ] Code is instrumented to report unexpected failures or increases in the failure rate
* [ ] Commits have been squashed so that each one has a clear purpose (and green tests)
* [ ] Proper labels have been applied

Problem
-------
1. Empty strings appeared in the editor (when only non-printable characters were included)
2. Spaces around the text where not preserved

Steps to reproduce
------------------
1. Use the ifood IDML file from this [doc](https://docs.google.com/document/d/1XEkdNjt5JCp-_shSTQ775L8rvnstrMA5-QiGsW1qHg4/edit)
2. Use the TrustYouBranding file from the same doc and the `recent research results` string

Solution
--------
1. Ignore strings that don't contain letters, symbols or punctuation (as defined [here](https://www.unicode.org/reports/tr44/#General_Category_Values))
2. Extend the regular expression to ignore wrapping spaces

### CODE REVIEW

File: openformats/formats/indesign.py [hunk 1]
```diff
@@ -104,6 +104,9 @@ def _can_skip_content(self, string):
             return True
         except ValueError:
             pass
+        # Special content in BackingStory.xml
+        if u'\ufeff' == string:
+            return True
```
Comment (kouk) [inline openformats/formats/indesign.py]: @rigaspapas why not just check for content after stripping all non-printable characters?
ref: https://mayart.de/download/Indesign-IDML/special-idml-chars.pdf

one idea would be to use the `unicodedata` module to ignore all character not in printable categories:
https://www.unicode.org/reports/tr44/#General_Category_Values

Comment (kouk) [reply openformats/formats/indesign.py]: The reason I'm saying this is who's to say we don't find another file that e.g. has a `u'\u200c'` character?

Comment (rigaspapas) [reply openformats/formats/indesign.py]: @kouk what we want to achieve is to avoid exporting text that is not translatable. There is no standard way in Unicode characters to distinguish letters from symbols. Is there any?

You are right that any non-printable character can cause the same unwanted result. I considered ignoring one specific character, because I guess that's what InDesign insert in every document automatically (not a user input).

File: openformats/tests/formats/indesign/test_indesign.py [hunk 2]
```diff
@@ -44,16 +44,22 @@ def test_can_skip_content(self):
...
-            '   <?ACE 7?> ',
-            '',
-            '<?ACE 8?> <Br/>;',
+            u' ',
+            u'   <?ACE 7?> ',
+            u'',
+            u'<?ACE 8?> <Br/>;',
+            u'\ufeff',
+            u' \ufeff ',
+            u' \ufeff 5',
```
Comment (kouk) [inline openformats/tests/formats/indesign/test_indesign.py]: here's another test case: 
```
u'<Br/>\ufeff'
```
;-)

File: openformats/formats/indesign.py [hunk 3]
```diff
@@ -104,6 +107,22 @@ def _can_skip_content(self, string):
...
+        translated. We assume that translatable characters are the letters,
+        the symbols and the punctuation.
+        """
+        acceptable = ["L", "P", "S"]
+
+        for letter in string:
+            char_type = unicodedata.category(letter)
+            if char_type[0] in acceptable:
+                return True
         return False
```
Comment (kouk) [inline openformats/formats/indesign.py]: just for fun:
```
from six.moves import map
return any(c[0] in ["L", "P", "S"] for c in map(unicodedata.category, string))
```
or even
```
from six.moves import map
return any(map(["L", "P", "S"].__contains__,
               map(itemgetter(0), map(unicodedata.category, string))) 
```

File: openformats/tests/formats/indesign/test_indesign.py (hunk 2 above)
Comment (rigaspapas) [reply openformats/tests/formats/indesign/test_indesign.py]: You are right. We should first strip the special characters and then check for the translatable ones. Thanks!

File: openformats/formats/indesign.py (hunk 3 above)
Comment (rigaspapas) [reply openformats/formats/indesign.py]: @kouk I really like the first approach, which is very *functional*, but it would require a new dependency. So, should we leave it as is?

Comment (kouk) [reply openformats/formats/indesign.py]: sure, it's fine like it is. But these examples don't really require the new dependency, you could do `from itertools import imap`. It's just that this way it's compatible with both python2 and python3.
